src/app/components/game-components/game-component/GameComponent.spec.ts

```diff
@@ -147,37 +115,43 @@ describe('GameComponent', () => {
             Yinsh: { onClick: [0, 0] },
         };
         const refusal: MGPValidation = MGPValidation.failure(GameWrapperMessages.NO_CLONING_FEATURE());
-
         for (const gameName of gameList.concat('MinimaxTesting')) {
-            const game: { [methodName: string]: unknown[] } = clickableMethods[gameName];
+            const game : { [methodName: string]: unknown[] } = clickableMethods[gameName];
             if (game == null) {
                 throw new Error('Please define ' + gameName + ' clickable method in here to test them.');
             }
             activatedRouteStub.setRoute('compo', gameName);
-            fixture = TestBed.createComponent(LocalGameWrapperComponent);
-            component = fixture.debugElement.componentInstance;
-            component.observerRole = 2;
-            fixture.detectChanges();
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            const testUtils: ComponentTestUtils<any> =
+                await ComponentTestUtils.forGame(gameName, LocalGameWrapperComponent, false);
+            const component: AbstractGameComponent = testUtils.getComponent();
+            testUtils.wrapper.observerRole = 2;
+            testUtils.detectChanges();
             tick(1);
-            expect(component.gameComponent).toBeDefined();
+            expect(component).toBeDefined();
             for (const methodName of Object.keys(game)) {
-                expect(component.gameComponent[methodName]).withContext(`click method ${methodName} should be defined for game ${gameName}`).toBeDefined();
-                const clickResult: MGPValidation = await component.gameComponent[methodName](...game[methodName]);
+                expect(component[methodName]).withContext(`click method ${methodName} should be defined for game ${gameName}`).toBeDefined();
+                const clickResult: MGPValidation = await component[methodName](...game[methodName]);
                 expect(clickResult).toEqual(refusal);
             }
         }
         tick(3000); // needs to be >2999
     }));
-    it('Component should have an encoder and a tutorial', fakeAsync(async() =>{
+    it('should have an encoder and a tutorial for every game', fakeAsync(async() =>{
         for (const gameInfo of GameInfo.ALL_GAMES()) {
             if (gameInfo.display === false) {
                 continue;
             }
-            const gameComponent: AbstractGameComponent =
-                TestBed.createComponent(gameInfo.component).debugElement.componentInstance;
-            expect(gameComponent.encoder).withContext('Encoder missing for ' + gameInfo.urlName).toBeTruthy();
-            expect(gameComponent.tutorial).withContext('tutorial missing for ' + gameInfo.urlName).toBeTruthy();
-            expect(gameComponent.tutorial.length).withContext('tutorial empty for ' + gameInfo.urlName).toBeGreaterThan(0);
+            activatedRouteStub.setRoute('compo', gameInfo.urlName);
```

Comment:
Pti GWT des familles ?
Si ce spec.ts s'allonge ça va devenir intéréssant de voir à faire un unique for avec plusieurs it imbriqués dedans ^^ (pour l'instant ça va)

Reply:
Fait ! Mais oui :D (ou dans le expectToBeCreated qui deviendrai expectToBeCreatedAndHaveTheExpectedDefinitions ou autre)